server/src/test/java/org/elasticsearch/common/util/BigArraysTests.java

```diff
@@ -389,6 +390,13 @@ public void testEstimatedBytesSameAsActualBytes() throws Exception {
         }
     }
 
+    public void testOverSizeUsesMinPageCount() {
+        final int pageSize = 1 << (randomIntBetween(2, 16));
+        final int minSize = randomIntBetween(1, pageSize) * randomIntBetween(1, 100);
+        final long size = BigArrays.overSize(minSize, pageSize, 1);
+        assertThat(size - minSize, lessThan((long) pageSize));
```

Comment:
Stronger assertions that would have avoided my confusion here:

```suggestion
        assertThat(size, greaterThanOrEqualTo((long)minSize));
        if (size >= pageSize) {
            assertThat(size + " is a multiple of " + pageSize, size % pageSize, equalTo(0L));
        }
        assertThat(size - minSize, lessThan((long) pageSize));
```

Reply:
++ makes sense